### PR TITLE
xmlsectool 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by gvarisco**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/xmlsectool.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Xmlsectool < Formula
+  desc "Download, check well-formedness, schema validity, and signature of an XML document. It can also create enveloped signatures of an XML document."
```
Comment (bfontaine) [inline Formula/xmlsectool.rb]: Please shorten this description; it should fit in 80 characters.

File: Formula/xmlsectool.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  homepage "https://wiki.shibboleth.net/confluence/display/XSTJ2/xmlsectool+V2+Home"
+  url "https://shibboleth.net/downloads/tools/xmlsectool/latest/xmlsectool-2.0.0-bin.zip"
```
Comment (bfontaine) [inline Formula/xmlsectool.rb]: We need to build from source here.

File: Formula/xmlsectool.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 "83ea1e977d947dd5f2f5600a3280bc5d372a4913f0dbce1ddf2c0ca1c869ca57"
+
+  depends_on :java => "1.7+"
+
+  def install
+    # Remove doc folder
+    rm_rf Dir["doc"]
+    # Install files
```
Comment (bfontaine) [inline Formula/xmlsectool.rb]: Please remove these comments.

File: Formula/xmlsectool.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    # temp workaround
+    inreplace "#{libexec}/xmlsectool.sh", "LOCATION=$0", "LOCATION=#{prefix}/libexec/"
+    bin.install_symlink "#{libexec}/xmlsectool.sh" => "xmlsectool"
+  end
+
+  def caveats; <<-EOS.undent
+   You may need to set JAVA_HOME:
+    export JAVA_HOME="$(/usr/libexec/java_home)"
+    EOS
+  end
```
Comment (bfontaine) [inline Formula/xmlsectool.rb]: This doesn’t seem Homebrew-specific.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/xmlsectool.rb (hunk 1 above)
Comment (gvarisco) [inline Formula/xmlsectool.rb]: I saw the jenkins' output. Fixed.

File: Formula/xmlsectool.rb (hunk 3 above)
Comment (gvarisco) [inline Formula/xmlsectool.rb]: I saw the jenkins' output. Fixed.

And comments removed.

File: Formula/xmlsectool.rb (hunk 4 above)
Comment (gvarisco) [inline Formula/xmlsectool.rb]: Removed. I added it as seen in other formulas. sorry! :)

**Comment by gvarisco** [comment]
@bfontaine thanks for the review. In order to build it from source ( git://git.shibboleth.net/xmlsectool.git ), what about the other dependencies (JARs)? How shall we build them? I'm talking about:
`└── libexec
    ├── lib
    │   ├── bcprov-jdk15on-1.53.jar
    │   ├── commons-codec-1.10.jar
    │   ├── cryptacular-1.0.jar
    │   ├── guava-18.0.jar
    │   ├── httpclient-4.3.6.jar
    │   ├── httpcore-4.3.3.jar
    │   ├── java-support-7.2.0.jar
    │   ├── jcl-over-slf4j-1.7.12.jar
    │   ├── jcommander-1.48.jar
    │   ├── joda-time-2.9.jar
    │   ├── jsr305-3.0.1.jar
    │   ├── logback-classic-1.1.3.jar
    │   ├── logback-core-1.1.3.jar
    │   ├── opensaml-core-3.2.0.jar
    │   ├── opensaml-messaging-api-3.2.0.jar
    │   ├── opensaml-security-api-3.2.0.jar
    │   ├── opensaml-security-impl-3.2.0.jar
    │   ├── opensaml-xmlsec-api-3.2.0.jar
    │   ├── opensaml-xmlsec-impl-3.2.0.jar
    │   ├── slf4j-api-1.7.12.jar
    │   ├── stax-api-1.0-2.jar
    │   ├── stax2-api-3.1.4.jar
    │   ├── woodstox-core-asl-4.4.1.jar
    │   ├── xmlsec-2.0.5.jar
    │   └── xmlsectool-2.0.0.jar`

**Comment by MikeMcQuaid** [comment]
@gvarisco Java doesn't need to be built from source but any native code does. This looks like it should be fine.

**Review by bfontaine — APPROVED** [review]

**Comment by gvarisco** [comment]
awesome, thanks @bfontaine and @MikeMcQuaid for the prompt review! We should be good to go, I believe.

File: Formula/xmlsectool.rb (hunk 1 above)
Comment (bfontaine) [inline Formula/xmlsectool.rb]: Seems like it’s still too long :)
```
Error: 1 problem in 1 formula
xmlsectool:
  * Description is too long. "name: desc" should be less than 80 characters.
    Length is calculated as xmlsectool + desc. (currently 85)
```

**Comment by bfontaine** [comment]
The test is failing: https://bot.brew.sh/job/Homebrew%20Core/19607/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_xmlsectool/

```
==> /usr/local/Cellar/xmlsectool/2.0.0/bin/xmlsectool --listBlacklist
ERROR: JAVA_HOME environment variable is not set.
Error: xmlsectool: failed
Failed executing: /usr/local/Cellar/xmlsectool/2.0.0/bin/xmlsectool --listBlacklist
...
```

File: Formula/xmlsectool.rb (hunk 1 above)
Comment (gvarisco) [inline Formula/xmlsectool.rb]: haha, I've missed the "xmlsectool + desc" part.. Fixing it now.

**Comment by gvarisco** [comment]
hey @bfontaine - saw it. I probably need your advice here :) 

"**/usr/local/Cellar/xmlsectool/2.0.0/bin/xmlsectool --listBlacklist**" works correctly once JAVA_HOME is set. 

`
$  /usr/local/Cellar/xmlsectool/2.0.0/bin/xmlsectool --listBlacklist
Digest algorithm blacklist:
   http://www.w3.org/2000/09/xmldsig#sha1
   http://www.w3.org/2001/04/xmldsig-more#md5

Signature algorithm blacklist:
   http://www.w3.org/2000/09/xmldsig#rsa-sha1
   http://www.w3.org/2001/04/xmldsig-more#ecdsa-sha1
   http://www.w3.org/2001/04/xmldsig-more#rsa-md5
`

I've added `ENV["JAVA_HOME"] = `/usr/libexec/java_home`.chomp` to the tests, but not sure if it is enough (as it still fails on my side, unless I set the JAVA_HOME variable). Is there any preferred way to have it set correctly?

File: Formula/xmlsectool.rb (hunk 1 above)
Comment (gvarisco) [inline Formula/xmlsectool.rb]: should be OK now.

**Comment by MikeMcQuaid** [comment]
> Is there any preferred way to have it set correctly?

Write a wrapper script which sets the variable. `grep` to see usage of `write_env_script` and `env_script_all_files`

**Comment by gvarisco** [comment]
@MikeMcQuaid oh, thanks for the heads up! Should be good now.

File: Formula/xmlsectool.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class Xmlsectool < Formula
+  desc "Check schema validity and signature of an XML document."
+  homepage "https://wiki.shibboleth.net/confluence/display/XSTJ2/xmlsectool+V2+Home"
+  url "https://shibboleth.net/downloads/tools/xmlsectool/latest/xmlsectool-2.0.0-bin.zip"
+  sha256 "83ea1e977d947dd5f2f5600a3280bc5d372a4913f0dbce1ddf2c0ca1c869ca57"
+
+  depends_on :java => "1.7+"
+
+  def install
+    rm_rf "doc"
```
Comment (dunn) [inline Formula/xmlsectool.rb]: Why remove this, out of curiosity?

Comment (gvarisco) [inline Formula/xmlsectool.rb]: It only contains:
````
├── LICENSE.txt
├── xmlsectool.patch
└── xmlsectool.spec
```
nothing relevant to copy over.

Comment (dunn) [inline Formula/xmlsectool.rb]: Couldn't hurt to `prefix.install "doc/LICENSE.TXT"` (we have an `#install_metafiles` method that does it automatically if it's in the directory root).

Comment (gvarisco) [inline Formula/xmlsectool.rb]: @dunn OK! Just added it.

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by MikeMcQuaid** [comment]
(fixed up unneeded additional metafiles installation in e42971f32d)

**Pull request closed without merging**